src/ng/httpBackend.js

```diff
@@ -88,6 +88,20 @@ function createHttpBackend($browser, createXhr, $browserDefer, callbacks, rawDoc
       xhr.onerror = requestError;
       xhr.onabort = requestError;
 
+      if (eventHandlers) {
+        forEach(eventHandlers, function(value, key) {
+          if (key !== 'upload') {
+            xhr.addEventListener(key, value);
+          }
+        });
+
+        if (eventHandlers.upload) {
```

Comment:
Yes, but we end up doing `xhr.addEventListener(key, value)` for all of them.
I.e. I could write you example as:

```js
{
  someUploadEvent: ...,
  someOtherUploadEvent: ...
  upload: {
    someEvent: ...,
    someOtherEvent: ...,
  }
}

// or

{
  someEvent: ...,
  someOtherEvent: ...,
  someUploadEvent: ...,
  someOtherUploadEvent: ...
}
```

and the result would be the same.

So, I am wondering if there is any benefit in separating the two. Why not keep it flat ?